Topic/more words to help home

prko opened the pull request:
## Purpose and Motivation
I recently compared the help document home pages (or index pages) of SC, Max, pd and Csound, and I found SC's documentation to be the most beginner unfriendly. The following three posts in a thread are relevant to this:
1. https://scsynth.org/t/developing-new-supercollider-tutorials/364/47?u=prko
2. https://scsynth.org/t/developing-new-supercollider-tutorials/364/49?u=prko
3. https://scsynth.org/t/developing-new-supercollider-tutorials/364/50?u=prko

Then I thought we needed more words for the help browser home page, and posted the following thread, where I also compared the SuperCollider 3.4 for Windows (I do not have a Mac to launch SC 3.4 on.) documentation with the current one:
https://scsynth.org/t/more-words-to-supercollider-help-document-home/8492?u=prko

This is the latest version, reflecting all the opinions in the thread above as best I can. I have also changed some links to be more appropriate. You can see these changes in the browser home of this PR in the following video:
https://www.dropbox.com/scl/fi/5531ljvpp1iy9j7uuduq3/sc-documentation-home-draft-4.mov?rlkey=ax82kto7erkn4dm2x056gvl2z&dl=0

As you can see in the video, the SuperCollider on the top left hides the S in the SuperCollider label. This is reported in https://github.com/supercollider/supercollider/issues/6138.

We should also update scdoc.css: the most important thing for this PR is to fix the width of the tables, which is also included in this PR. 

The scdoc.js should also be modified to include the class tree in the indexes menu at the top. This is also included in this PR. 

### Notes

1. To arrange the table more beautifully, I am using ```soft:: ::``` for a space placeholder.
The character between :: and :: is an en space: 
`en space | 020002 | 8194 | 0x2002 | &ensp;`


2. I have not added a link for sclang, as I am not sure which is better for it between `Process`, `Main` and `Interpreter`.

3. Would it be better to show the paths of sclang, scsynth, supernova and scide per OS? Showing these executables in the file explorer seems to help beginners understand the inter-application cooperation.
4. I am not sure what the correct category is for 'Audio device selection'. Would "Basic" in "Get Help" be better than the current category ("Basic" in "Overview")?

## Types of changes

<!-- Delete lines that don't apply -->

- Documentation

## To-do list

<!-- Complete an item by checking it: [x]. Add new entries to track your progress -->
- [x] I hope that the source files included in this PR (and other files not included in this PR) will be modified by many users to get a better help documentation home for beginners, intermediate and advanced users, as well as for non-SC users who might read the help documentation home from the Internet out of interest. Feel free to change the details of this PR.
- [ ] Code is tested
- [x] All tests are passing
- [x] Updated documentation
- [x] This PR is ready for review
<img width="984" alt="Screenshot 2023-11-23 at 20 48 24" src="https://github.com/supercollider/supercollider/assets/416281/e21d2b25-d926-4880-9286-e949ce15be7c">
<img width="985" alt="Screenshot 2023-11-23 at 20 48 38" src="https://github.com/supercollider/supercollider/assets/416281/c925c824-1690-4383-abf5-2207996b9b95">

prko commented:
I hope that the source files included in this PR (and other files not included in this PR) will be modified by many users to get a better help documentation home for beginners, intermediate and advanced users, as well as for non-SC users who might read the help documentation home from the Internet out of interest. Feel free to change the details of this PR.

capital-G commented on HelpSource/Help.schelp:
```diff
@@ -2,38 +2,266 @@ TITLE::Help
+
+SECTION:: An introductory overview
+
+SuperCollider* is a cross-platform** environment for audio synthesis and algorithmic composition used by musicians, artists and researchers working with sound, with three main components:
```
```suggestion
*SuperCollider* is a cross-platform environment for audio synthesis and algorithmic composition used by musicians, artists and researchers working with sound, with three main components:
```

prko replied:
Thanks! However, I have no experience with this kind of work. 
 
I pressed "commit suggestion", and then it seemed that your suggestion was not applied. So I pressed "Resolve conversation". 

What should I do to apply your suggestion? It is not easy to understand the documentation on GitHub.

prko replied:
Or does someone else apply the suggestion? Sorry for these questions.. (:

prko replied:
@capital-G 

Thank you very much! However, I have no experience with this type of work.
I clicked on 'Commit suggestion' and then it seemed that your suggestion was not being applied. So I clicked on 'Resolve conversation'.

What should I do to get your suggestion committed? The documentation on GitHub is not easy to understand.
Or is someone else committing it?

I will look into how the process works. If your suggestion is ignored, it is my fault. Please let me know if there is anything I can do to get it committed.

mtmccrea replied:
This was new for me as well.
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request

prko replied:
@mtmccrea 
Thanks!

@capital-G 
I think you mean italic style by adding two * symbols around SuperCollider? I changed it to `emphasis::SuperCollider::`. Please let me know if I am wrong.

Yes, updating scdoc.css might be better. I will try again with minimal changes.

mtmccrea replied:
> I think you mean italic

There may have been a misunderstanding of the original intention of the asterisks being for markdown (bold), when actually those were meant as footnote indicators.

capital-G commented on HelpSource/Help.schelp:
```diff
@@ -2,38 +2,266 @@ TITLE::Help
+
+link::Guides/ClientVsServer##The server and client:: communicate via link::http://opensoundcontrol.org/##OSC (open sound control):: messages over link::Guides/Glossary#kw_udp#UDP::, allowing SuperCollider to run on a single machine or link::Guides/MultiClient_Setups##on multiple machines over a network::. Users can also diretly control the sc-server using link::Guides/NodeMessaging::.
+
+If you wish, you can extend SuperCollider's functionality by installing optional link::https://github.com/supercollider/sc3-plugins##sc3-plugins:: distributed by the SC development team, and by installing individual extensions from the custom libraries called link::Classes/Quarks:: developed by individual users. For more information, see the link::https://github.com/supercollider-quarks/quarks##D ...
```
```suggestion
If you wish, you can extend SuperColliders functionality by installing optional link::https://github.com/supercollider/sc3-plugins##sc3-plugins:: distributed by the SC development team, and by installing individual extensions from the custom libraries called link::Classes/Quarks:: developed by individual users. For more information, see the link::https://github.com/supercollider-quarks/quarks##Directory of community contributed Quarks for SuperCollider:: and link::https://baryon.supercollider.online/##Baryon::, a package index for SuperCollider Quarks.
```

capital-G replied:
quarks.supercollider.online is deprecated in favor of baryon :)

mtmccrea replied:
"SuperCollider's" was correct

capital-G replied:
my bad, sorry :D

capital-G commented on HelpSource/Help.schelp:
```diff
@@ -2,38 +2,266 @@ TITLE::Help
+link::https://supercollider.github.io/downloads.html#mac#macOS:: and
+link::https://supercollider.github.io/downloads.html#windows#Windows::,
+as well as microcomputers such as
+link::https://github.com/supercollider/supercollider/blob/develop/README_RASPBERRY_PI.md##Raspberry Pie:: and
```
```suggestion
link::https://github.com/supercollider/supercollider/blob/develop/README_RASPBERRY_PI.md##RaspberryPi:: and
```

capital-G commented on HelpSource/Help.schelp:
```diff
@@ -2,38 +2,266 @@ TITLE::Help
+link::https://github.com/supercollider/supercollider/blob/develop/README_BEAGLEBONE_BLACK.md##BeagleBone Black::.
+link::https://www.arduino.cc/##Arduino:: or equivalent devices can be
+controlled via link::Classes/SerialPort:: within sclang.
+DMX can also be used by means of user-developed libraries (link::https://quarks.supercollider.online/##type dmx in the search box on the linked web page::
```
```suggestion
DMX can also be used by means of user-developed libraries.
```

capital-G reviewed (commented):
Some minor corrections/suggestions from my side :)

mtmccrea commented on HelpSource/Help.schelp:
```diff
@@ -2,38 +2,266 @@ TITLE::Help
+
+## soft:: ::• soft:: ::Client soft:: ::– sclang. a text-based programming language and interpreter. ||
+
+emphasis::sclang:: controls audio synthesis processes on the server and offers an link::Overviews/JITLib##interactive programming environment for on-the-fly coding::, link::Tutorials/A-Practical-Guide/PG_01_Introduction##an extensive library for creating patterns:: and link::Classes/Scale##a library of scales and modes from various cultures::.
```
```suggestion
emphasis::sclang:: controls audio synthesis processes on the server and offers an interactive programming environment for link::Overviews/JITLib##on-the-fly coding::, an extensive library for creating link::Tutorials/A-Practical-Guide/PG_01_Introduction##patterns:: and a library of link::Classes/Scale##scales and modes:: from various cultures.
```

mtmccrea replied:
I'd also suggest "live coding" instead of "on-the-fly" coding, just because it's a bit more recognizable these days.

prko replied:
> I'd also suggest "live coding" instead of "on-the-fly" coding, just because it's a bit more recognizable these days.

done!

mtmccrea commented on HelpSource/Help.schelp:
```diff
@@ -2,38 +2,266 @@ TITLE::Help
+## soft:: ::• soft:: ::Client soft:: ::– sclang. a text-based programming language and interpreter. ||
+
+emphasis::sclang:: controls audio synthesis processes on the server and offers an link::Overviews/JITLib##interactive programming environment for on-the-fly coding::, link::Tutorials/A-Practical-Guide/PG_01_Introduction##an extensive library for creating patterns:: and link::Classes/Scale##a library of scales and modes from various cultures::.
+link::Tutorials/Getting-Started/15-Sequencing-with-Routines-and-Tasks##Creating processes that unfold over time:: is straighforward in sclang.
+It also provides link::Guides/GUI-Introduction##all the essential features of a graphical user interface:: that can be used to create custom user interfaces and data visualization.
+With support for
+link::Guides/UsingMIDI##MIDI::,
+link::Guides/OSC_communication##OSC:: and
+link::Guides/Working_with_HID##HID::,
+sclang can be easily interfaced with local and networked hardware and software.
```
```suggestion
Creating link::Tutorials/Getting-Started/15-Sequencing-with-Routines-and-Tasks##processes:: that unfold over time is straightforward in emphasis::sclang::.
It also provides all the essential features of a graphical user interface (link::Guides/GUI-Introduction##GUI::) that can be used to create custom user interfaces and data visualization.
With support for
link::Guides/UsingMIDI##MIDI::,
link::Guides/OSC_communication##OSC:: and
link::Guides/Working_with_HID##HID::,
emphasis::sclang:: can be easily interfaced with local and networked hardware and software.
```

mtmccrea commented on HelpSource/Help.schelp:
```diff
@@ -2,38 +2,266 @@ TITLE::Help
+## soft:: ::• soft:: ::link::Classes/Server:: – a real-time audio engine.||
+The default server is link::Classes/Server#Switching the server application#scsynth::. To take full advantage of multi-core, an alternative sound engine, link::Classes/Server#Switching the server application#supernova::, can be used.
+
+## soft:: ::• soft:: ::Client soft:: ::– sclang. a text-based programming language and interpreter. ||
```
```suggestion
## soft:: ::• soft:: ::Client soft:: ::– sclang, a text-based programming language and interpreter. ||
```

mtmccrea commented on HelpSource/Help.schelp:
```diff
@@ -2,38 +2,266 @@ TITLE::Help
+sclang can be easily interfaced with local and networked hardware and software.
+
+## soft:: ::• soft:: ::link::Classes/ScIDE:: soft:: ::– a dedicated editor for SuperCollider with an integrated help system.||
+link::Guides/SCIde##The SuperCollider-IDE (Integrated Development Environment) :: allows you to start writing code right out of the box. Community members also maintain sclang integration with other popular IDEs such as link::https://github.com/scztt/vscode-supercollider##Visual Studio Code, VSCodium::, link::https://github.com/capital-G/sc_kernel##Jupyter::, link::https://github.com/davidgranstr ...
```
```suggestion
The SuperCollider Integrated Development Environment (link::Guides/SCIde##IDE::) allows you to start writing code right out of the box. Community members also maintain sclang integration with other popular IDEs such as link::https://github.com/scztt/vscode-supercollider##Visual Studio Code, VSCodium::, link::https://github.com/capital-G/sc_kernel##Jupyter::, link::https://github.com/davidgranstrom/scnvim##SCNvim::, link::https://github.com/supercollider/scvim##scvim::, link::https://github.com/supercollider/scel##Emacs::, and so on.
```

mtmccrea replied:
I think it’s ok to list alternative IDEs, but probably best to avoid linking to repos outside of SC's GitHub organization. It would be best to link to a wiki on the SC repo that has a list of IDEs with SC support. That way it can be maintained more dynamically without the need to update the help docs (which can take 6+ months between updates).

prko replied:
> It would be best to link to a wiki on the SC repo that has a list of IDEs with SC support.

this is a very good idea! I could not find a page on a wiki. Where can I find it?

cdbzb replied:
well there's this: https://github.com/supercollider/supercollider/wiki/Systems-interfacing-with-SC - needs some editing perhaps!

prko replied:
> https://github.com/supercollider/supercollider/wiki/Systems-interfacing-with-SC

Ah yes!  However, the Jupyter extension and the VSCode extension should be added to this page.

I can link that page. Then the following sentence
>Community members also maintain sclang integration with other popular IDEs such as link::https://github.com/scztt/vscode-supercollider##Visual Studio Code, VSCodium::, link::https://github.com/capital-G/sc_kernel##Jupyter::, link::https://github.com/davidgranstrom/scnvim##SCNvim::, link::https://github.com/supercollider/scvim##scvim::, link::https://github.com/supercollider/scel##Emacs::, and so on.

should be changed as follows?

>Community members also maintain link::https://github.com/supercollider/supercollider/wiki/Systems-interfacing-with-SC##sclang integration with other systems such as various programming languages and editors.::

mtmccrea commented on HelpSource/Help.schelp:
```diff
@@ -2,38 +2,266 @@ TITLE::Help
+sclang can be easily interfaced with local and networked hardware and software.
+
+## soft:: ::• soft:: ::link::Classes/ScIDE:: soft:: ::– a dedicated editor for SuperCollider with an integrated help system.||
+link::Guides/SCIde##The SuperCollider-IDE (Integrated Development Environment) :: allows you to start writing code right out of the box. Community members also maintain sclang integration with other popular IDEs such as link::https://github.com/scztt/vscode-supercollider##Visual Studio Code, VSCodium::, link::https://github.com/capital-G/sc_kernel##Jupyter::, link::https://github.com/davidgranstr ...
```
> other popular IDEs such as 

I think it’s ok to list alternative IDEs, but probably best to avoid linking to repos outside of SC's GitHub organization. It would be best to link to a wiki on the SC repo that has a list of IDEs with SC support. That way it can be maintained more dynamically without the need to update the help docs (which can take 6+ months between updates).

mtmccrea commented on HelpSource/Help.schelp:
```diff
@@ -2,38 +2,266 @@ TITLE::Help
+link::Guides/SCIde##The SuperCollider-IDE (Integrated Development Environment) :: allows you to start writing code right out of the box. Community members also maintain sclang integration with other popular IDEs such as link::https://github.com/scztt/vscode-supercollider##Visual Studio Code, VSCodium::, link::https://github.com/capital-G/sc_kernel##Jupyter::, link::https://github.com/davidgranstr ...
+::
+
+link::Guides/ClientVsServer##The server and client:: communicate via link::http://opensoundcontrol.org/##OSC (open sound control):: messages over link::Guides/Glossary#kw_udp#UDP::, allowing SuperCollider to run on a single machine or link::Guides/MultiClient_Setups##on multiple machines over a network::. Users can also diretly control the sc-server using link::Guides/NodeMessaging::.
```
```suggestion
The server and client link::Guides/ClientVsServer##communicate:: via Open Sound Control (link::http://opensoundcontrol.org/##OSC::), allowing SuperCollider to run on a single machine or on link::Guides/MultiClient_Setups##multiple machines:: over a network. Users can also control the audio server directly with any OSC-enabled program by using the link::Guides/NodeMessaging:: protocol.
```

mtmccrea replied:
I omitted the part about UDP because the interested user can get that info deeper in the docs or the linked OSC page, and also because TCP can also be enabled (though this is uncommon).

mtmccrea commented on HelpSource/Help.schelp:
```diff
@@ -2,38 +2,266 @@ TITLE::Help
+
+link::Guides/ClientVsServer##The server and client:: communicate via link::http://opensoundcontrol.org/##OSC (open sound control):: messages over link::Guides/Glossary#kw_udp#UDP::, allowing SuperCollider to run on a single machine or link::Guides/MultiClient_Setups##on multiple machines over a network::. Users can also diretly control the sc-server using link::Guides/NodeMessaging::.
+
+If you wish, you can extend SuperCollider's functionality by installing optional link::https://github.com/supercollider/sc3-plugins##sc3-plugins:: distributed by the SC development team, and by installing individual extensions from the custom libraries called link::Classes/Quarks:: developed by individual users. For more information, see the link::https://github.com/supercollider-quarks/quarks##D ...
```
```suggestion
An extensive library of Unit Generators—the building blocks of audio synthesis networks—are included with SuperCollider. Users may optionally extend this core library with the link::https://github.com/supercollider/sc3-plugins##sc3-plugins:: suite of UGens. Additionally, there are many of user-developed language extensions, called link::Classes/Quarks::, to further extend the functionality of SuperCollider. Quarks can be browsed in the link::https://baryon.supercollider.online/##Baryon:: index, or in the distribution link::https://github.com/supercollider-quarks/quarks##repository:: where users can submit their own Quarks for distribution.
```

prko replied:
Sorry, I committed the change from the oldest suggestions. Thus, this change is outdated. Please understand that I am working on this for the first time. I updated it manually.

mtmccrea commented on HelpSource/Help.schelp:
```diff
@@ -2,38 +2,266 @@ TITLE::Help
+as well as microcomputers such as
+link::https://github.com/supercollider/supercollider/blob/develop/README_RASPBERRY_PI.md##Raspberry Pie:: and
+link::https://github.com/supercollider/supercollider/blob/develop/README_BEAGLEBONE_BLACK.md##BeagleBone Black::.
+link::https://www.arduino.cc/##Arduino:: or equivalent devices can be
+controlled via link::Classes/SerialPort:: within sclang.
+DMX can also be used by means of user-developed libraries (link::https://quarks.supercollider.online/##type dmx in the search box on the linked web page::
```
I don't think that Arduino and DMX should be mentioned in a footnote about _cross-platform_ capabilities.

I suggested adding SerialPort to the protocol list in the Client section. Note sure where/if to mention Arduino and DMX elsewhere, or if those are best left to other docs.

mtmccrea replied:
Also, I'm not sure about using the table format for local footnotes. Just using the footnotes tag may be better, even though it will put them at the bottom of the page. 
I could see the argument for the supported platforms to be listed as another paragraph in this introduction, rather than a footnote.

prko replied:
I removed the sentences mentioning Arduino and DMX. I also removed tables for local footnotes and inserted those paragraphs into the first paragraph of the introduction.

mtmccrea commented:
One final thought after this first read-through:
I nearly always have the Help pane docked in the IDE, occupying about 1/4 of the window’s width. Can you post a screenshot of the tables with a narrow pane width?

prko commented:
> Can you post a screenshot of the tables with a narrow pane width?

Yes! Here you are:
<img width="2056" alt="Screenshot 2023-11-26 at 9 30 37" src="https://github.com/supercollider/supercollider/assets/416281/38a7f979-8786-4ed7-b03f-dd039dbf73cb">
<img width="539" alt="Screenshot 2023-11-26 at 9 30 49" src="https://github.com/supercollider/supercollider/assets/416281/01ea0f0e-d925-4223-b338-4669ff74e09c"><img width="539" alt="Screenshot 2023-11-26 at 9 31 08" src="https://github.com/supercollider/supercollider/assets/416281/d8d72a85-648b-4e6d-96f9-c33f67866388"><img width="538" alt="Screenshot 2023-11-26 at 9 31 27" src="https://github.com/supercollider/supercollider/assets/416281/a8ff50dc-1019-42dd-8b30-c517b3d66732"><img width="540" alt="Screenshot 2023-11-26 at 9 32 02" src="https://github.com/supercollider/supercollider/assets/416281/4a22028a-9b2f-4183-a625-fb5c6a059699"><img width="540" alt="Screenshot 2023-11-26 at 9 33 38" src="https://github.com/supercollider/supercollider/assets/416281/91e90730-2f45-4523-90b8-34776d3157a6">

prko commented:
I changed the left margin of the unordered list in table td in scdoc.css.
I also removed soft::    :: which was used to adjust the width of the table cells, but not removed soft:: :: used for other purposes.

The following screenshots are taken with the minimum width of the Help Browser to display the Help Home correctly:

<img width="2056" alt="Screenshot 2023-11-26 at 10 40 31" src="https://github.com/supercollider/supercollider/assets/416281/932786df-b4b3-4054-8c21-c34f4da75650">
<img width="693" alt="Screenshot 2023-11-26 at 10 41 11" src="https://github.com/supercollider/supercollider/assets/416281/9e341c61-b402-4835-9244-da030a77e38b">
<img width="694" alt="Screenshot 2023-11-26 at 10 41 45" src="https://github.com/supercollider/supercollider/assets/416281/27cb647f-dc41-415f-8330-ae576974eb30">
<img width="694" alt="Screenshot 2023-11-26 at 10 42 00" src="https://github.com/supercollider/supercollider/assets/416281/94091bfb-65c1-4293-b781-557fc77d4a8e">
<img width="694" alt="Screenshot 2023-11-26 at 10 42 15" src="https://github.com/supercollider/supercollider/assets/416281/31e0553c-a2c4-4977-beb5-f1bd1c400272">

prko commented:
I have split the "Contributing to SC" section into "Sharing your work" and "Contributing to SC". 
I changed the left margin of the unordered list in table td in scdoc.css. As far as I have checked, there seems to be no list in a table in the default installation of SuperCollider. The "Help.schelp" file is the only unique help file.
I also removed soft::    :: which was used to adjust the width of the table cells, but not removed soft:: :: used for other purposes.

The following screenshots are taken with the minimum width of the Help Browser to display the Help Home correctly:
<img width="2056" alt="Screenshot 2023-11-26 at 13 46 25" src="https://github.com/supercollider/supercollider/assets/416281/ea96ecd1-c707-484c-9673-17d876eab2d3">
<img width="697" alt="Screenshot 2023-11-26 at 13 46 42" src="https://github.com/supercollider/supercollider/assets/416281/89d6aecf-bb6f-43ee-b1be-d1e120cf0f32">
<img width="694" alt="Screenshot 2023-11-26 at 13 46 59" src="https://github.com/supercollider/supercollider/assets/416281/7ae6811a-4f76-4d56-a0d9-dc2ea9642f33">

<img width="695" alt="Screenshot 2023-11-26 at 13 47 19" src="https://github.com/supercollider/supercollider/assets/416281/70bb12ee-2cd2-4867-844d-25d892c5286e">

prko commented:
I have split the "Contributing to SC" section into "Sharing your work" and "Contributing to SC". 
I changed the left margin of the unordered list in table td in scdoc.css. As far as I have checked, there seems to be no list in a table in the default installation of SuperCollider. The "Help.schelp" file is the only unique help file.
I also removed soft::    :: which was used to adjust the width of the table cells, but not removed soft:: :: used for other purposes.

The following screenshots are taken with the minimum width of the Help Browser to display the Help Home correctly:
<img width="2056" alt="Screenshot 2023-11-26 at 15 01 44" src="https://github.com/supercollider/supercollider/assets/416281/2105c02d-4830-4deb-860b-2ded8d650a6e">
<img width="696" alt="Screenshot 2023-11-26 at 15 02 02" src="https://github.com/supercollider/supercollider/assets/416281/a47d13fe-bdcd-4f81-bba7-4fb8536f55ed">
<img width="693" alt="Screenshot 2023-11-26 at 15 02 21" src="https://github.com/supercollider/supercollider/assets/416281/a711bf21-fa7b-482d-9f7e-f9e450c50be8">
<img width="693" alt="Screenshot 2023-11-26 at 15 02 38" src="https://github.com/supercollider/supercollider/assets/416281/6b6c6b59-5b8e-4f92-82e6-b417433817f0">
<img width="693" alt="Screenshot 2023-11-26 at 15 02 56" src="https://github.com/supercollider/supercollider/assets/416281/c84cf244-94b7-4560-ad35-a2dae5cc2c53">

prko commented:
I have split the "Contributing to SC" section into "Sharing your work" and "Contributing to SC". 
I changed the left margin of the unordered list in table td in scdoc.css. As far as I have checked, there seems to be no list in a table in the default installation of SuperCollider. The "Help.schelp" file is the only unique help file.
I also removed soft::    :: which was used to adjust the width of the table cells, but not removed soft:: :: used for other purposes.
The following screenshots are taken with the minimum width of the Help Browser to display the Help Home correctly:

<img width="2056" alt="Screenshot 2023-11-26 at 15 07 48" src="https://github.com/supercollider/supercollider/assets/416281/6bd4cec5-1ec6-4dc4-98a6-8601dc23dba7">
<img width="693" alt="Screenshot 2023-11-26 at 15 08 11" src="https://github.com/supercollider/supercollider/assets/416281/90a694a8-bc14-45d8-802c-b59a8d6b0505">
<img width="694" alt="Screenshot 2023-11-26 at 15 08 37" src="https://github.com/supercollider/supercollider/assets/416281/b47950b6-277b-4cf4-b09e-6b0dd1f794ef">
<img width="693" alt="Screenshot 2023-11-26 at 15 08 53" src="https://github.com/supercollider/supercollider/assets/416281/ee9b6e51-6182-4ae5-8a2a-8ec8f68743b7">
<img width="693" alt="Screenshot 2023-11-26 at 15 09 08" src="https://github.com/supercollider/supercollider/assets/416281/7d454887-34e5-4859-9487-02b38ff3da8e">

mtmccrea commented:
Thanks @prko for the screenshots and making the adjustments to improve the appearance for narrower views!

capital-G commented:
@prko maybe it would be best to use css grid layouts for the tables instead? Currently the width of each column is not aligned.

<img width="833" alt="image" src="https://github.com/supercollider/supercollider/assets/8267062/e2f59afe-d654-4600-993b-6a57d4f48a6b">


Using grid view allows in an easy way to distribute content equally as well as changing the layout easily on really narrow views (e.g. 200px) through e.g.

```css
@media only screen and (max-width: 200px) {
    .table-content {
        grid-template-columns: 1fr;
    }
}
```

If you need some help for this I am happy to help :)

prko commented:
@capital-G 

> @prko maybe it would be best to use css grid layouts for the tables instead? Currently the width of each column is not aligned.
> ...
> If you need some help for this I am happy to help :)

This is a great idea. I found the following website:
https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_grid_layout/Basic_concepts_of_grid_layout

To implement this functionality, a class (according to the page the class "wrapper") should be used in the `scdoc.css` file and the `help.schelp` file, if I understand it correctly.

Am I correct?

If I am correct, I can put the following part in `scdoc.css`:
```
.wrapper {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
}
```
However, I do not know how to put its corresponding counterpart `<div class="wrapper">` into the `Help.schelp` file. The `SCDoc.sc` file or/and the `SCDocRenderer.sc` file should seem to be modified, but this is beyond my capabilities. 

So I need your help and I would be very grateful if you could help me!

prko commented:
@capital-G 
> The `SCDoc.sc` file or/and the `SCDocRenderer.sc` file should seem to be modified,

Some tags seem to be related to `.cpp', `.hpp`, `.l`, `.y` files which are used to build a package of SC: 
https://github.com/supercollider/supercollider/issues/6131

prko commented:
the table design is changed. seems better?
<img width="498" alt="Screenshot 2024-01-05 at 16 21 46" src="https://github.com/supercollider/supercollider/assets/416281/9eee319b-3355-4300-b3f6-3453c248f9dd">

mtmccrea commented on HelpSource/Help.schelp:
```diff
@@ -2,38 +2,328 @@ TITLE::Help
+
+SECTION:: An introductory overview
+
+emphasis::SuperCollider:: is a cross-platform environment for audio synthesis and algorithmic composition used by musicians, artists and researchers working with sound. It can be installed and run on link::https://supercollider.github.io/downloads.html#linux#Linux::,
```
Maybe just link to the downloads page by the word "installed" rather than linking to each individual OS download.

prko replied:
Done, thanks!

mtmccrea commented on HelpSource/Help.schelp:
```diff
@@ -2,38 +2,328 @@ TITLE::Help
+The server and client link::Guides/ClientVsServer##communicate:: via Open Sound Control (link::http://opensoundcontrol.org/##OSC::
+), allowing SuperCollider to run on a single machine or on link::Guides/MultiClient_Setups##multiple machines:: over a network. Users can also control the audio server directly with any OSC-enabled program by using the link::Guides/NodeMessaging:: protocol.
+
+An extensive library of Unit Generators—the building blocks of audio synthesis networks—are included with SuperCollider. Users may optionally extend this core library with the link::https://github.com/supercollider/sc3-plugins##sc3-plugins:: suite of UGens. Additionally, there are many of user-developed language extensions, called link::Classes/Quarks::, to further extend the functionality of Sup ...
```
I don't think Baryon has been agreed on as an official quark directory.

prko replied:
Thanks! Baryon is now removed.

prko commented:
**The current design (the smallest width)**.
The even cell width results in a wider page width.
<img width="721" alt="Screenshot 2024-02-03 at 17 28 23" src="https://github.com/supercollider/supercollider/assets/416281/57ede91b-cc76-40ec-aa78-c15d242be1b2">
<img width="722" alt="Screenshot 2024-02-03 at 17 28 50" src="https://github.com/supercollider/supercollider/assets/416281/16728a9d-96a3-4dd1-aa0e-73d20b18fd6e">
<img width="722" alt="Screenshot 2024-02-03 at 17 29 07" src="https://github.com/supercollider/supercollider/assets/416281/8ff492f2-88db-47b5-a7de-14fba80774d2">
![Screenshot 2024-02-03 at 17 32 49](https://github.com/supercollider/supercollider/assets/416281/1b19ed64-facf-4290-938d-326a7706ec60)
<img width="2056" alt="Screenshot 2024-02-03 at 17 32 00" src="https://github.com/supercollider/supercollider/assets/416281/06a22a41-43c0-4262-abb1-e864bbc318d6">

prko commented:
This seems to be the last change from my point of view, but I am willing to change if anyone asks for a change. 

- Vertical cell width arrangement is better (still not perfect).
- Horizontal width is now the most compact.
- Links in Contribute and Share have been added and redesigned.

Here is some screenshots on Windows:
![Screenshot 2024-02-09 at 21 20 40](https://github.com/supercollider/supercollider/assets/416281/91cea791-3409-4844-ab9d-fbdd8bdefe7a)
![Screenshot 2024-02-09 at 21 20 22](https://github.com/supercollider/supercollider/assets/416281/58a28c65-dd05-4f20-a566-75c99865ae75)

@capital-G 
I would like to know if you have a solution for the following: 

> Using grid view allows in an easy way to distribute content equally as well as changing the layout easily on really narrow views (e.g. 200px) through e.g.
> 
> ```css
> @media only screen and (max-width: 200px) {
>     .table-content {
>         grid-template-columns: 1fr;
>     }
> }
> ```
> 
> If you need some help for this I am happy to help :)

As I wrote earlier, I cannot find a way to do this myself, unfortunately...

capital-G commented:
I just checked out the code and the problem is that scdoc is not really following best practices regarding DOM construction, so it is not possible to give a namespace to the described tables and therefore adjust only their styling. It also seems not possible to construct a div namespace "on the fly" within the document.

One can create a css selector along "select the table after the heading of title xyz" but this is prone to fail - additionally the individual blocks are not wrapped in a div and the individual blocks are also not divs, which makes alignment really hacky.

I tried to create a similar flow via `definitionlist::` but it allows no nesting of lists or linebreaks, so I instead tried to use `definitionlist` with a `list` which looks like

<img width="536" alt="image" src="https://github.com/supercollider/supercollider/assets/8267062/7f573906-d74d-4c7c-9054-68ea3a18e076">

```supercollider
definitionlist::
     ## Basics
     || List::
          ## link::Guides/How-to-Use-the-Interpreter::
          ## link::Guides/Understanding-Errors::
     ::
     ## Intermediate
     || List::
          ## link::Guides/Debugging-tips::
          ## link::Guides/Tracing-Processes::
          ## link::Guides/HID_permissions::
          ## link::Guides/LID_permissions::
     ::
     ## Intermediate
     || List::
          ## link::Guides/Internal-Snooping::
     ::
::
```

While I looked at it I also noticed some strange symbols in the code which should not be there - I think you use them to create spacing, but this is more a hack and one should use CSS instead, but this doesn't work either as stated above.

<img width="606" alt="image" src="https://github.com/supercollider/supercollider/assets/8267062/11640c27-ccec-4ae8-b717-d2e4c58380ed">

Generally, scdoc would benefit from a rewrite implementing div namespaces which would allow for improved styling capabilities, but this would be out of scope.

I think its best to use the existing design patterns within scdoc, as this makes a future re-design (or else) easier if there are not too different elements or hacks present.

prko commented:
For me, the design is not that important. I am more satisfied with the current content because
- I can do more things in the help browser, and
- I think I can make SuperCollider more beginner-friendly.

I hope one of this one or the previous one could be accepted with the content based on the current content.

Here is the current state:

![Screenshot 2024-02-10 at 0 54 29](https://github.com/supercollider/supercollider/assets/416281/2b1c4117-9851-4a76-862d-76b617b126bb)

![Screenshot 2024-02-10 at 0 54 38](https://github.com/supercollider/supercollider/assets/416281/af133ec3-8bd3-425f-8201-be665b7a6e20)

![Screenshot 2024-02-10 at 0 54 44](https://github.com/supercollider/supercollider/assets/416281/48fb72eb-9574-42fe-9cd9-60eb770de76c)

![Screenshot 2024-02-10 at 0 55 14](https://github.com/supercollider/supercollider/assets/416281/d845b2ff-01dd-40a5-816e-26e41781cd83)

![Screenshot 2024-02-10 at 0 55 19](https://github.com/supercollider/supercollider/assets/416281/62a50d95-14a2-4ccc-a08b-2c3ba25cb827)

![Screenshot 2024-02-10 at 0 55 23](https://github.com/supercollider/supercollider/assets/416281/bb69f369-e716-4d5f-8490-a45aa81e3de2)

prko commented:
https://github.com/supercollider/supercollider/pull/6265#issuecomment-2115709830
@capital-G I think the design of the help home document could be redesigned using your idea!

dyfer commented:
Hi @prko, what's the status of this PR?

prko commented:
This is the end state from my perspective. What do @mtmccrea, @capital-G and @cdbzb think?

prko commented:
I just submitted the last commit from my perspectives:
- restoring scdoc.css
- change a linked text length
 
This is the end state from my perspective. What do @mtmccrea, @capital-G and @cdbzb think?

capital-G commented:
One thing which I don't know how to solve: It really depends on ones background what is advanced and what is beginner level. I think the SC docs in particular are always hesitant to assume background of the reader, which creates a wild mix of documentation style compared to e.g. Max - but I personally welcome the more _opinionated approach_ b/c it also provides "more experienced readers" an overview and people w/o such experience are provided multiple entry paths, something which the docs could improve and I think this PR does exactly that.
I really enjoy that some really interesting bits are immediately seen by the user when they start - it creates the feeling of a TOC in a book.

I think there could be a bigger "visible emphasis" on the basic concepts such as `SynthDef` & `Synth`, `Ndef`, `Tdef/Task`, `Pdef/Pbind` and `Bus` b/c I think once people tackle these basic structures they tackled the gist of SC :) Although the explanation of those should be handled in a tutorial it wouldn't hurt if these basic classes would show up in some of your tables IMO :)

Regarding the Client/Server architecture: I recently re-wrote the introduction text on the website https://supercollider.github.io/ where I tried my spin on the server/client architecture - it doesn't need to replace yours, but I always struggled to understand how *server* and *audio* get along, because when I hear server I either used to think of a webserver or a physical server.

Generally I think @tedmoore should give some guidance on how to move forward here :)

prko commented:
>One thing which I don't know how to solve: It really depends on ones background what is advanced and what is beginner level. 

The categorised levels are for beginners. Intermediate and advanced users do not need such category. 
However, I removed these categories and numbereated those items.

I reuploaded the commit reflecting @capital-G 's opinions and new writings in https://supercollider.github.io/.
![Screenshot 2025-04-14 at 19 15 16](https://github.com/user-attachments/assets/39d9a13e-1065-42b5-b8e5-c14429f30ee1)
![Screenshot 2025-04-14 at 19 15 27](https://github.com/user-attachments/assets/8cff654f-d1b0-43fc-b04e-7b305973a728)
![Screenshot 2025-04-14 at 19 15 38](https://github.com/user-attachments/assets/114bf3e1-205b-46bd-8a4f-f2dc715949c2)
![Screenshot 2025-04-14 at 19 15 48](https://github.com/user-attachments/assets/fe7b2b73-45a6-439d-9f23-557567763a0c)
![Screenshot 2025-04-14 at 19 16 02](https://github.com/user-attachments/assets/cbc7826d-6a8a-4c73-b96a-fd0ede5e4308)
![Screenshot 2025-04-14 at 19 16 14](https://github.com/user-attachments/assets/99628482-0855-4556-93d5-f48f863deedd)

prko commented:
Looking back, this PR initially had enthusiastic involvement from some users, and I made all the changes according to their requests. However, it's truly unfortunate that this PR remains in its current state. Still, with so many unmarked PRs left behind, there's hardly room to express my regret here. It's quite saddening…

tedmoore commented:
Hi @prko, I know I've been asked to look at it. It's on my to-do list.

mtmccrea commented:
Thanks for the review @tedmoore.

@capital-G, would you mind performing the (squash) merge, as you have been tracking the development closely? (@cdbzb feel free also to offer any final thoughts).

I haven't been following super closely, but in general this seems like a nice improvement and provides a useful structure that can easily accommodate future tweaks to which specific topics are highlighted on the home page.

Thanks!
